aziotctl/aziotctl-common/src/system/status.rs

```diff
@@ -8,140 +8,152 @@ use anyhow::{Context, Result};
-        for result in results {
-            if let Status::Failed(_) = result.service_status {
-                print_logs(result.service_name, &result.service_status)?;
-            }
-            for socket in result.sockets {
-                if !socket.ok() {
-                    print_logs(socket.socket_name, &result.service_status)?;
-                }
-            }
+        if !matches!(service.state(), State::Failed(_)) {
+            continue;
         }
 
-        let name = crate::program_name();
-        println!();
-        println!();
-        println!("Note: inactive services with active sockets are considered OK, while inactive sockets or services with no sockets to start them are considered failed. This is because services can be inactive if not in use.");
-        println!("For more detailed logs, use the `{} system logs` command. If the logs do not contain enough information, consider setting debug logs using `{} system set-log-level`.", name, name);
-    } else {
-        println!("Ok");
+        println!(
+            "{} is in a bad state because:",
+            service
+                .name
+                .strip_suffix(".service")
+                .unwrap_or(service.name)
+        );
+
+        print_logs(service.name, &service.state)?;
+        for socket in service.sockets {
+            if !matches!(socket.state, State::Active) {
+                print_logs(socket.name, &socket.state)?;
+            }
+        }
     }
 
-    Ok(())
-}
-
-fn read_status(process: &str) -> Result<Status> {
-    let result = Command::new("systemctl")
-        .args(&["is-active", process])
-        .output()
-        .context("Failed to call systemctl is-active")?;
-
-    let output = String::from_utf8_lossy(&result.stdout);
-
-    let result = match output.trim() {
-        "active" => Status::Active,
-        "inactive" => Status::Inactive,
-        _ => Status::Failed(format!(
-            "{} {}",
-            String::from_utf8_lossy(&result.stdout),
-            String::from_utf8_lossy(&result.stderr)
-        )),
-    };
-
-    Ok(result)
-}
-
-fn print_logs(process: &str, state: &Status) -> Result<()> {
+    let name = crate::program_name();
+    println!("Use '{} system logs' to check for non-fatal errors.", name);
     println!(
-        "{} is in a bad state: {:?}. Printing the last 10 log lines.",
-        process, state,
+        "Use '{} check' to diagnose connectivity and configuration issues.",
+        name
     );
-    Command::new("journalctl")
-        .args(&["-u", process, "--no-pager", "-e", "-n", "10"])
-        .spawn()
-        .context("Failed to spawn new process for printing logs")?
-        .wait()
-        .context("Failed to call journalctl")?;
-    println!();
 
     Ok(())
 }
 
-#[derive(PartialEq, Debug)]
-enum Status {
+#[derive(Clone, Debug)]
+enum State {
     Active,
-    Failed(String),
     Inactive,
+    Failed(String),
+}
+
+impl State {
+    fn from_systemctl(process: &str) -> Result<State> {
```

Comment:
`s/process/unit/`

Reply:
sure, I was just matching lee's terminology, but unit is more appropriate